### PR TITLE
6. As a user, I want to be able to invite others to an existing shopping list #6

### LINKED ISSUE #6: 6. As a user, I want to be able to invite others to an existing shopping list
## Summary

We want to allow users to invite others to existing shopping lists so they can manage them with friends or family. This would take the following steps:

1. User enters an existing user's email into a form
2. The `shareList` function in `src/api/firebase.js` is called with the appropriate inputs

Please note that an existing user is a user that has signed into the app before.

## Acceptance criteria

- [x]  The `ManageList` view shows a form that allows the user to enter an email to invite an _existing_ user to a list, in addition to the form that allows them to add items to that list.
- [x]  The input that accepts the email  has a semantic `label` element associated with it
- [x]  The user can submit this form with both the mouse and the `Enter` key
- [x]  If the other user exists, the user is alerted that the list was shared
- [x]  If the other user _does not_ exist, the user is shown an error message that explains the problem

## Notes

- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- Check the `shareList` function in `src/api/firebase.js` to get an idea of how you will know if a list was shared.

### PR DESCRIPTION
**Pull request by eonflower**
## Description
- Added modal component for future modal-ing (missing styling for true modal effect)
- Added InviteForm component that allows user to input an email for sharing a list
- Added button to ManageList view that conditionally renders "modal" with the InviteForm component
- Updated shareList function in firebase.js to render alerts based on the user inputted email

Something that we thought about, but didn't delve into, is having an alert for if the user we are trying to invite already has the list we are trying to share. It looked too in depth for the scope of this issue.

We chose to utilize a modal because we ran into the issue of not being able to submit on enter with two forms. The modal will still need styling :)

## Related Issue

closes #6 

## Acceptance Criteria

- [x]  The `ManageList` view shows a form that allows the user to enter an email to invite an _existing_ user to a list, in addition to the form that allows them to add items to that list.
- [x]  The input that accepts the email  has a semantic `label` element associated with it
- [x]  The user can submit this form with both the mouse and the `Enter` key
- [x]  If the other user exists, the user is alerted that the list was shared
- [x]  If the other user _does not_ exist, the user is shown an error message that explains the problem

## Type of Changes

enhancement

### Before

tcl-66-issue-6-Before

### After

tcl-66-issue-6-After-1
tcl-66-issue-6-After-2

## Testing Steps / QA Criteria

- checkout an-sc-issue-6 
- git pull
- npm start
- from home page, click the list you want to manage/share
- navigate to manage-list page
- click invite button
- enter the email of the user you want to invite
- press enter/return or click invite
- after closing the alert, the modal should close
- can go to firebase (https://console.firebase.google.com/u/3/project/tcl-66-smart-shopping-list/firestore/data/~2F) to verify that list has been shared

notes: 
- if user does not exist, an error will pop up
- if user does exist, a success message will pop up
- if you are not owner of the list, an error will pop up

### CODE REVIEW

File: src/components/InviteForm.jsx
Comment (hun-ah) [inline src/components/InviteForm.jsx]: Would using just a string as the value of input make more sense here? (see next comment for more details)

File: src/components/InviteForm.jsx
Comment (hun-ah) [inline src/components/InviteForm.jsx]: If you use a string instead of an object you could just do `setInput(e.target.value)` and save a few lines of code. Wondering if using an object doesn't make the most sense since there is only a single value and this could be a bit verbose for what you're trying to achieve? Just a thought!

**Review by hun-ah — APPROVED** [review]
Looks good overall - nice work! I like the "modal"; adding styling in the future will be great. See comments re: string vs. object for input state.

**Review by 3campos — APPROVED** [review]
I reviewed your code and it looks good! I ran it locally and received the expected messages for a successful share attempt and an unsuccessful share attempt. I'll leave it up to you to decide how you'd like to incorporate @hun-ah's feedback. You've got my approval though. Nice work!

File: src/components/InviteForm.jsx
Comment (eonflower) [reply src/components/InviteForm.jsx]: We made it this way because of the parameters for the shareList function and wanting to keep the same naming for the firebase schema. We initially had just an empty string, but it wasn't working properly. Though,  we also changed some other things. Could try it out again.

Comment (eonflower) [reply src/components/InviteForm.jsx]: Okay, I tested the code with these changes and it still worked. Thanks for the input!

File: src/components/Modal.jsx
Comment (jeremiahfallin) [inline src/components/Modal.jsx:1]: Really liking the Modal component! My only suggestion would be to consider the dialog html element:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

File: src/api/firebase.js
Comment (jeremiahfallin) [inline src/api/firebase.js]: Shouldn't need this console.log anymore.

File: src/api/firebase.js
Comment (jeremiahfallin) [inline src/api/firebase.js:1]: I think it makes more sense to return a string from this function and then do the alert on the "frontend." This works for this project but if this were a serverless function you wouldn't be able to call the alert.

**Review by jeremiahfallin — COMMENTED** [review]
Leaving some notes. Will follow up with you and if you don't want to make any changes I'll approve the PR.

File: src/components/Modal.jsx
Comment (eonflower) [reply src/components/Modal.jsx:1]: I think the way Stefie and I were imagining it would be to have it eventually be an overlayed modal that would pop up. We chose not to style it for now, but that was our intention. To my knowledge, the dialog element is more for inline interaction?? But I also have not used it very much.

Comment (StefieCaff) [reply src/components/Modal.jsx:1]: Thanks for sharing that @jeremiahfallin I had not read much about that html element until now.

**Review by jeremiahfallin — APPROVED** [review]
Looks good to me!

**Pull request merged**
